wremnants/include/csVariables.h

```diff
@@ -76,5 +78,30 @@ CSVars csSineCosThetaPhi(const PtEtaPhiMVector &antilepton, const PtEtaPhiMVecto
     return angles;
 }
 
+CSVars csSineCosThetaPhiTransported(const PtEtaPhiMVector &antilepton, const PtEtaPhiMVector &lepton, const PtEtaPhiMVector &targetV) {
```

Comment:
What is different about this?

Reply:
This is needed because pythia doesn't store the lepton kinematics for the intermediate states, so to compute the CS variables, we compute the lepton momenta in the rest frame of the W/Z with the nominal preFSR kinematics, and then boost back to the alternative rest from of the W/Z corresponding to the intermediate pythia state (this is what's provided by the targetV argument)